### PR TITLE
Update user controllers

### PR DESCRIPTION
**Pull request by vitaliipp**
Such user controllers have been updated:
getUserDetails, updatePassword, updateProfile, getAllUsers, getSingleUser

### CODE REVIEW

**Review by Nata-Kalina — APPROVED** [review]
LGTM. Tested getUserDetails, updatePassword, updateProfile, getAllUsers, getSingleUser controllers. All of them works fine.

File: src/controllers/userController.js [hunk 1]
```diff
@@ -7,66 +7,63 @@ const {
 } = require('../errors');
 
 const getUserDetails = async (req, res, next) => {
-  const user = await User.findById(req.params.id);
-  if (!user) {
-    throw new NotFoundError(`No user with id ${req.params.id}`);
-  } else {
-    res.status(StatusCodes.OK).json({ user });
-  }
+  res.status(StatusCodes.OK).json({ user: req.user });
```
Comment (OxiBo) [inline src/controllers/userController.js:10]: Your req.user object only contains the following properties: name, userId and role which you send back in the response. I suspect the front end would like to see more information about the user, like an email or even favorite pets list. I would discuss this with your front end team. The version of the similar route you had before included more information which frontend team was using on their end.

File: src/controllers/userController.js [hunk 2]
```diff
@@ -7,66 +7,63 @@ const {
...
-      throw new UnauthenticatedError('Password does not match');
-    }
-    user.password = req.body.newPassword;
-    await user.save();
-    res.status(StatusCodes.OK).json({
-      msg: `Password for user ${user.name} was successfully updated `,
-    });
+  const { oldPassword, newPassword, confirmPassword } = req.body;
+  console.log(oldPassword, newPassword, confirmPassword);
+  if (!oldPassword || !newPassword || !confirmPassword) {
```
Comment (OxiBo) [inline src/controllers/userController.js:16]: Usually it is client side concern if passwords match, but if you do go with this approach, you should let your frontend team that they need to send both values.

File: src/controllers/userController.js [hunk 3]
```diff
@@ -7,66 +7,63 @@ const {
...
+
+  const token = user.createJWT();
+  res
+    .status(StatusCodes.OK)
+    .cookie('token', token, {
+      expires: new Date(Date.now() + 60 * 24 * 3600000),
+      httpOnly: true,
+      signed: true,
+    })
+    .json({ user: { name: user.name }, token, userId: user._id });
```
Comment (OxiBo) [inline src/controllers/userController.js:53]: I think in one of your latest PRs you decided to send more information about the user - like user email. You might want to talk to your frontend team about what information about the user they need back when updating user profile.

**Review by OxiBo — COMMENTED** [review]
Good job overall. The PR could be merged but I added a few comments most of which are about communicating with the frontend team to find out which information they want back in the responses. Please do discuss those questions with them.

**Comment by vitaliipp** [comment]
Thank you for your review. I greatly appreciate your input and I am actively working on incorporating the necessary corrections in the upcoming updates.

**Pull request merged**